src/Tools/CsvImporter/Program.cs

```diff
@@ -1,251 +1,116 @@
-using EntraMfaPrefillinator.Lib.Models;
-using EntraMfaPrefillinator.Lib.Services;
+﻿using System.Text.Json;
+using EntraMfaPrefillinator.Tools.CsvImporter;
+using EntraMfaPrefillinator.Tools.CsvImporter.Extensions;
+using EntraMfaPrefillinator.Tools.CsvImporter.Extensions.QueueClient;
+using EntraMfaPrefillinator.Tools.CsvImporter.Logging;
 using EntraMfaPrefillinator.Tools.CsvImporter.Models;
+using EntraMfaPrefillinator.Tools.CsvImporter.Services;
 using EntraMfaPrefillinator.Tools.CsvImporter.Utilities;
+using Microsoft.Extensions.Azure;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+string configDirPath = Path.Combine(
+    path1: Environment.IsPrivilegedProcess
+        ? Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData)
+        : Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData),
+    path2: "EntraMfaPrefillinator",
+    path3: "CsvImporter"
+);
 
-Stopwatch stopwatch = Stopwatch.StartNew();
-
-// Set Azure Storage connection string and dry run flag from environment variables.
+string logsDirPath = Path.Combine(
+    path1: configDirPath,
+    path2: "Logs"
+);
 
-// Set CSV file path and max tasks from command line arguments.
-string csvFilePathArg;
-try
-{
-    csvFilePathArg = args[0];
-}
-catch (Exception)
+if (!Directory.Exists(configDirPath))
 {
-    throw new ArgumentException("Missing CSV file path argument");
+    Directory.CreateDirectory(configDirPath);
 }
 
-string maxTasksArg;
-try
+if (!Directory.Exists(logsDirPath))
 {
-    maxTasksArg = args[1];
-}
-catch (Exception)
-{
-    maxTasksArg = "256";
+    Directory.CreateDirectory(logsDirPath);
 }
 
-int maxTasks;
-try
-{
-    maxTasks = int.Parse(maxTasksArg);
-}
-catch (Exception ex)
-{
-    throw new ArgumentException($"Invalid max tasks argument: {maxTasksArg}", ex);
-}
+string configFilePath = Path.Combine(configDirPath, "config.json");
+string logFilePath = Path.Combine(logsDirPath, $"CsvImporter_{DateTime.UtcNow:yyyy-MM-dd_HH-mm-ss}.log");
 
-// Read config file.
-CsvImporterConfig csvImporterConfig;
-try
-{
-    csvImporterConfig = await ConfigFileUtils.GetCsvImporterConfigAsync();
-}
-catch (Exception ex)
-{
-    throw new Exception($"Error reading config file: {ex.Message}");
-}
+bool configFileExists = File.Exists(configFilePath);
 
-QueueClientService queueClientService;
-try
+CsvImporterConfig? preLaunchConfig = null;
+if (configFileExists)
 {
-    string storageConnectionString = Environment.GetEnvironmentVariable("STORAGE_CONNECTION_STRING") ?? throw new NullReferenceException("STORAGE_CONNECTION_STRING environment variable not set");
-
-    queueClientService = new(
-        connectionString: storageConnectionString
+    preLaunchConfig = JsonSerializer.Deserialize(
+        json: File.ReadAllText(configFilePath),
+        jsonTypeInfo: CoreJsonContext.Default.CsvImporterConfig
     );
 }
```

Comment:
Need to look into figuring out a better way of handling this. I'll probably switch to adding it to the generic host config.